packages/react-select/src/components/Select/useSelect.ts

```diff
@@ -0,0 +1,38 @@
+import * as React from 'react';
+import { getNativeElementProps, resolveShorthand } from '@fluentui/react-utilities';
+import type { SelectProps, SelectSlots, SelectState } from './Select.types';
+
+/**
+ * Array of all shorthand properties listed as the keys of SelectSlots
+ */
+export const selectShorthandProps: (keyof SelectSlots)[] = ['select', 'icon', 'root'];
+
+/**
+ * Create the state required to render Select.
+ *
+ * The returned state can be modified with hooks such as useSelectStyles,
+ * before being passed to renderSelect.
+ *
+ * @param props - props from this instance of Select
+ * @param ref - reference to root HTMLSelectElement of Select
+ */
+export const useSelect = (props: SelectProps, ref: React.Ref<HTMLElement>): SelectState => {
+  const { select, icon, size, appearance, inline } = props;
+
+  return {
+    size,
+    appearance,
+    inline,
+    components: {
+      root: 'span',
+      select: 'select',
+      icon: 'span',
+    },
+    select: resolveShorthand(select, { required: true }),
```

Comment:
As I see `disabled` is defined in `SelectProps`, is it intentional that it's not passed down to `select`?

If not:

```suggestion
    select: resolveShorthand(select, { defaultProps: { disabled }, required: true }),
```

Wider question: I would except that `Select` will follow "primary slot" pattern (sample implementation #20863). Do you want to address this in current PR or in a separate?

Reply:
I'd planned to update that in a separate PR (actually on my todo list)

Reply:
I updated this PR to have `select` be the primary slot 👍